lib/rex/proto/dns/resolver.rb

```diff
@@ -84,8 +84,15 @@ def initialize(config = {})
           raise ResolverArgumentError, "Option #{key} not valid"
         end
       end
+
       self.static_hostnames = StaticHostnames.new(hostnames: static_hosts)
-      self.static_hostnames.parse_hosts_file
+      begin
+        self.static_hostnames.parse_hosts_file
+      rescue StandardError => e
+        @logger.error 'Failed to parse the hosts file, ignoring it'
```

Comment:
Looks like the default `@logger` is `/dev/null`:

```
[11] pry(#<Rex::Proto::DNS::CachedResolver>)> @config[:log_file]
=> "/dev/null"
```

Which in this scenario might not be useful if folk want to debug issues/report framework issues; Is it possible to wire it up to the framework logging